packages/strapi-admin/admin/src/containers/Admin/index.js

```diff
@@ -0,0 +1,386 @@
+    const { global: { plugins } } = props;
+
+    return !Object.keys(plugins).every(plugin => (plugins[plugin].isReady === true));
+  } 
+
+  helpers = {
+    hideLeftMenu: this.props.hideLeftMenu,
+    hideLogout: this.props.hideLogout,
+    setAppSecured: this.props.setAppSecured,
+    showLeftMenu: this.props.showLeftMenu,
+    showLogout: this.props.showLogout,
+    unsetAppSecured: this.props.unsetAppSecured,
+    updatePlugin: this.props.updatePlugin,
+  };
+
+  isAcceptingTracking = () => {
+    const { admin: { uuid } } = this.props;
+
+    return !!uuid;
+  }
+
+  /**
+   * Display the app loader until the app is ready
+   * @returns {Boolean}
+   */
+  showLoader = () => {
+    const {
+      admin: { isLoading },
+      global: { isAppLoading },
+    } = this.props;
+
+    if (isAppLoading) {
+      return true;
+    }
+
+    if (isLoading) {
+      return true;
+    }
+
+    return this.hasApluginNotReady(this.props);
+  }
+
+  renderInitializers = () => {
+    const {
+      global: { plugins },
+    } = this.props;
+
+    return Object.keys(plugins).reduce((acc, current) => {
+      const Compo = plugins[current].initializer;
+      const key = plugins[current].id;
+
+      if (Compo) {
+        // We don't check if the initializer is correct because there's a fallback in cdc
+        acc.push(<Compo key={key} {...this.props} {...this.helpers} />);
+      }
+
+      return acc;
+    }, []);
+  };
+
+  renderMarketPlace = props => <Marketplace {...props} {...this.props} />;
+
+  renderPluginDispatcher = props => {
+    // NOTE: Send the needed props instead of everything...
+
+    return <PluginDispatcher {...this.props} {...props} {...this.helpers} />;
+  }
+
+  render() {
+    const {
+      admin: {
+        appError,
+        isLoading,
+        layout,
+        showLogoutComponent,
+        showMenu,
+        strapiVersion,
+      },
+      global: {
+        blockApp,
+        overlayBlockerData,
+        plugins,
+        showGlobalAppBlocker,
+      },
+    } = this.props;
+
+    if (appError) {
+      return <div>An error has occured please check your logs</div>;
+    }
+
+    if (isLoading) {
+      return <LoadingIndicatorPage />;
+    }
+
+    // We need the admin data in order to make the initializers work
+    if (this.showLoader()) {
+      return (
+        <React.Fragment>
+          {this.renderInitializers()}
+          <LoadingIndicatorPage />
+        </React.Fragment>
+      );
+    }
+
+    return (
+      <div className={styles.adminPage}>
+        {showMenu  && (
+          <LeftMenu
+            layout={layout}
+            version={strapiVersion}
+            plugins={plugins}
+          />
+        )}
+        <NavTopRightWrapper>
+          {/* Injection zone not ready yet */}
+          {showLogoutComponent && <Logout />}
+          <LocaleToggle isLogged />
+        </NavTopRightWrapper>
+        <div className={styles.adminPageRightWrapper} style={this.getContentWrapperStyle().main}>
```

Comment:
`this.getContentWrapperStyle().main` and `this.getContentWrapperStyle().sub`, I pretty sure there is a more convenient way to do it.

Reply:
Sure with a constant but it was harder to test than with an instance.

Reply:
I would prefer to see `this.getContentWrapperStyle('main')`